src/crypto/signature.js

```diff
@@ -57,10 +57,15 @@ export function parseSignatureParams(algo, signature) {
     }
     // Algorithm-Specific Fields for Ed25519 signatures:
     // - 64 octets of the native signature
-    case enums.publicKey.ed25519: {
-      const RS = signature.subarray(read, read + 64); read += RS.length;
+    // Algorithm-Specific Fields for Ed448 signatures:
+    // - 114 octets of the native signature
+    case enums.publicKey.ed25519:
+    case enums.publicKey.ed448: {
+      const rsSize = algo === enums.publicKey.ed25519 ? 64 : 114;
```

Comment:
Since you have the function anyway, you could do:
```suggestion
      const rsSize = 2 * getPayloadSize(algo);
```